### PR TITLE
Added temporary Railways TFC compat

### PR DESCRIPTION
**Pull request by pietro-lopes**
- Locked dependency to this exact railways version
- Cursed Enum Mixin ([by gigaherz](https://github.com/gigaherz/JsonThings/blob/master/src/main/java/dev/gigaherz/jsonthings/mixin/ResourcePackTypeExtender.java))

All this stuff can be removed when Steam and Rails add their compat.

### CODE REVIEW

File: src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        throw new IllegalStateException("Unreachable");
+    }
+
+    static
+    {
+        var entry = create("TFC", $VALUES.length, "tfc");
+
+        GregitasUtil.RailwaysTFC = entry;
+
+        //noinspection ShadowFinalModification
```
Comment (screret) [inline src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java:31]: hear me out: `@Mutable`.

File: build.gradle [hunk 2]
```diff
@@ -150,6 +153,11 @@ dependencies {
     modImplementation("dev.ftb.mods:ftb-library-forge:${project.ftb_library_version}") { transitive = false }
     modImplementation("dev.ftb.mods:ftb-teams-forge:${project.ftb_teams_version}") { transitive = false }
 
+    modRuntimeOnly("com.jozufozu.flywheel:flywheel-forge-1.20.1:0.6.10-7")
+    modRuntimeOnly("curse.maven:create-328085:4835191") { transitive = false }
+    modImplementation("blank:Steam_Rails-1.5.3+forge-mc1.20.1") // fixed Accessor
```
Comment (screret) [inline build.gradle:158]: do they not have a maven?

**Review by screret — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (screret) [reply build.gradle:158]: they do. `https://maven.ithundxr.dev/releases`

Comment (screret) [inline build.gradle:158]: use the create maven. located at `https://maven.tterrag.com/`

File: src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java (hunk 1 above)
Comment (pietro-lopes) [reply src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java:31]: feel free to do what is needed, I don't know what is that about, I just copy pasted
:monkey: see :monkey: do

File: build.gradle (hunk 2 above)
Comment (pietro-lopes) [reply build.gradle:158]: I was crashing when tried to use their mavem with "slim" version, didn't know they had an "all", this one does not crash :+1:

Comment (pietro-lopes) [reply build.gradle:164]: Added, still crashing but still added

File: src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java (hunk 1 above)
Comment (pietro-lopes) [reply src/main/java/com/allthemods/gravitas2/core/mixin/RailwaysModEnumMixin.java:33]: Added.

**Review by screret — APPROVED** [review]

**Comment by IThundxr** [comment]
@pietro-lopes could you grab a log of it crashing with the dep on my maven? I'll look into fixing that since people should be depending on the stable jars from the maven (release ones without build numbers) also this is so cursed & I love it lol

**Comment by pietro-lopes** [comment]
> @pietro-lopes could you grab a log of it crashing with the dep on my maven? I'll look into fixing that since people should be depending on the stable jars from the maven (release ones without build numbers) also this is so cursed & I love it lol

It is here:
https://github.com/Layers-of-Railways/Railway/blob/1.20/dev/common/src/main/java/com/railwayteam/railways/mixin/client/AccessorTrackBlockOutline.java

I added remap to false, and for what I know (little about mixin lol) remap should be false for everything it is **NOT** vanilla

About the error you can see here: https://github.com/architectury/architectury-mixin/issues/7

**Comment by IThundxr** [comment]
> > @pietro-lopes could you grab a log of it crashing with the dep on my maven? I'll look into fixing that since people should be depending on the stable jars from the maven (release ones without build numbers) also this is so cursed & I love it lol
> 
> It is here: https://github.com/Layers-of-Railways/Railway/blob/1.20/dev/common/src/main/java/com/railwayteam/railways/mixin/client/AccessorTrackBlockOutline.java
> 
> I added remap to false, and for what I know (little about mixin lol) remap should be false for everything it is **NOT** vanilla
> 
> About the error you can see here: [architectury/architectury-mixin#7](https://github.com/architectury/architectury-mixin/issues/7)

Hmm yeah that's weird, I'll try fixing that by defining the field the accessor is trying to get but I'm not sure why architectury is being like that though

**Comment by pietro-lopes** [comment]
> > > @pietro-lopes could you grab a log of it crashing with the dep on my maven? I'll look into fixing that since people should be depending on the stable jars from the maven (release ones without build numbers) also this is so cursed & I love it lol
> > 
> > 
> > It is here: https://github.com/Layers-of-Railways/Railway/blob/1.20/dev/common/src/main/java/com/railwayteam/railways/mixin/client/AccessorTrackBlockOutline.java
> > I added remap to false, and for what I know (little about mixin lol) remap should be false for everything it is **NOT** vanilla
> > About the error you can see here: [architectury/architectury-mixin#7](https://github.com/architectury/architectury-mixin/issues/7)
> 
> Hmm yeah that's weird, I'll try fixing that by defining the field the accessor is trying to get but I'm not sure why architectury is being like that though

You mean explicit define the field? I tried that and didn't work (didn't rename the getLONG... tho)

**Pull request merged**